apm-agent-premain/src/main/java/co/elastic/apm/agent/premain/AgentMain.java

```diff
@@ -130,12 +108,13 @@ public synchronized static void init(String agentArguments, Instrumentation inst
      * @return {@code true} for any Java 7 and early Java 8 HotSpot JVMs, {@code false} for all others
      */
     static boolean shouldDelayOnPremain() {
-        int majorVersion = JvmRuntimeInfo.getMajorVersion();
+        JvmRuntimeInfo runtimeInfo = JvmRuntimeInfo.ofCurrentVM();
+        int majorVersion = runtimeInfo.getMajorVersion();
         return
             (majorVersion == 7) ||
             // In case bootstrap checks were disabled
-            (majorVersion == 8 && JvmRuntimeInfo.isHpUx() && JvmRuntimeInfo.getUpdateVersion() < 2) ||
-            (majorVersion == 8 && JvmRuntimeInfo.isHotSpot() && JvmRuntimeInfo.getUpdateVersion() < 40);
+            (majorVersion == 8 && runtimeInfo.isHotSpot() && runtimeInfo.getUpdateVersion() < 2) ||
+            (majorVersion == 8 && runtimeInfo.isHotSpot() && runtimeInfo.getUpdateVersion() < 40);
```

Comment:
[question] so the current state here is that those JVM versions are not supported out of the box in `JavaVersionBootstrapCheck`, but just in case the bootstraps checks are disabled we still apply the startup delay.
Would it make sense to have all those per-version checks all done in `JavaVersionBootstrapCheck` so that everything that is related to JVM version is within the same class ? (Might be enough as static method though).